pkg/controllers/customroute/custom_route_conditions.go

```diff
@@ -15,105 +15,123 @@ import (
 	"github.com/openshift/cluster-authentication-operator/pkg/controllers/common"
 	"github.com/openshift/library-go/pkg/route/routeapihelpers"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/applyconfigurations/meta/v1"
 	corev1listers "k8s.io/client-go/listers/core/v1"
 )
 
-func ensureDefaultConditions(conditions []metav1.Condition) []metav1.Condition {
+var (
+	conditionStatusTrue      = metav1.ConditionTrue
+	conditionStatusFalse     = metav1.ConditionFalse
+	conditionTypeDegraded    = "Degraded"
+	conditionTypeProgressing = "Progressing"
+)
+
+func ensureDefaultConditions(conditions []*v1.ConditionApplyConfiguration) []*v1.ConditionApplyConfiguration {
```

Comment:
how did we start using SSA in 4.17?

Reply:
Note that this controller [is](https://github.com/openshift/cluster-authentication-operator/blob/release-4.17/pkg/controllers/customroute/custom_route_controller.go#L266) already using SSA in 4.17.

What changed is the signature of the `ComponentRouteStatusApplyConfiguration.WithConditions` method due to  https://github.com/openshift/client-go/pull/277

basically, before the change the method accepted:
```
(b *ComponentRouteStatusApplyConfiguration) WithConditions(values ...metav1.Condition) *ComponentRouteStatusApplyConfiguration
```

and after the change it accepts:
```
(b *ComponentRouteStatusApplyConfiguration) WithConditions(values ...*metav1.ConditionApplyConfiguration) *ComponentRouteStatusApplyConfiguration
```

The changes made in this PR also adapt the controller to use the new signature.